engine/combat/combat_controller.py

```diff
@@ -25,164 +22,57 @@
 
 
 class CombatController:
-    SLUG_TIMING = 0.35
-
     def __init__(self: Self) -> None:
-        self.reasoner = SoSReasoner()
-        self.action = None
-        self.ctrl = sos_ctrl()
-        self.block_timing = 0.0
-        self.second_attack = False
+        self.controller = None
 
     # returns a bool to feed to the sequencer
     def execute_combat(self: Self, delta: float) -> bool:
-        self.ctrl.set_neutral()
-
-        # if combat is done, just exit
-        if combat_manager.encounter_done is True:
-            return True
-
-        # if we are in the starting zone, just execute a non-timed basic attack unless
-        # we have already set the special action for the second encounter
-        # TODO(eein): Move these custom settings over to custom controllers after
-        # refactoring the controllers to use a base class and split methods.
-        # have it check if a state is valid, and set set a custom controller and
-        # execute on those actions, or use `self` to execute the default.
-        self._handle_encounter_controller()
-        # if some dialog is on the screen - make it go away
-        if new_dialog_manager.dialog_open:
-            self.ctrl.toggle_turbo(True)
-            self.ctrl.confirm()
-            self.ctrl.toggle_turbo(False)
-            self.second_attack = True
-            return False
-
-        # We need to decide how to handle these specific scenarios; via profile
-        # or whatever else, but this is good for now.
-        # Note: It can't be stopped or tested mid encounter.
+        sos_ctrl().set_neutral()
 
-        # if we dont have an action or the current appraisal is complete,
-        # we make a new one.
-        # we also check if battle command has focus, so it doesn't start executing before
-        # we have control
+        # Assign a controller, or the correct controller if it changes.
+        # This is because sometimes when we check, the controller has not been
+        # set by the game yet, or potentially a change in controllers during a fight.
```

Comment:
As mentioned in the chat earlier, some encounters like DoD might be somewhat funky and require a change in controllers.

Reply:
Dweller of Dread has a game-controller we can read from (DwellerOfDread = auto()) - we'll have to see what each fight shows for its controller and we can make unique ones if necessary - I think it should be straightforward to figure out if needed though :)

Reply:
Yeah, was referring to the clone sub-fights.